modules/framework/galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/common/RunResultUtility.java

```diff
@@ -20,85 +21,98 @@
 import dev.galasa.framework.spi.ResultArchiveStoreException;
 import dev.galasa.framework.spi.teststructure.TestMethod;
 import dev.galasa.framework.spi.teststructure.TestStructure;
+import dev.galasa.framework.api.common.Environment;
+import dev.galasa.framework.api.common.EnvironmentVariables;
 
 public class RunResultUtility {
-	
-	public static @NotNull RasRunResult toRunResult(@NotNull IRunResult runResult, boolean isMethodDetailsExcluded) throws ResultArchiveStoreException {
-	   
-	    String runId = runResult.getRunId();
-		RasTestStructure structure = getTestStructure(runResult);
+
+	Environment env;
+
+	public RunResultUtility(Environment env) {
+		this.env = env;
+	}
```

Comment:
We could actually add another constructor in here, so this block is like this:
```
Environment env;

    public RunResultUtility() {
		this(new SystemEnvironment());
	}

	public RunResultUtility(Environment env) {
		this.env = env;
	}
```

So the first constructor can be used by callers who don't care about passing in an environment/MockEnvironment and can just call `new RunResultUtility()` and don't have to worry if there's not an `env` object available to them there. But people who do care about the environment, like the unit tests, can call `new RunResultUtility(mockEnv)`.